### PR TITLE
Redesign to master

### PR DESCRIPTION
**Pull request by nati-rapaport**

### CODE REVIEW

**Review by ronen-haen-sndk — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -6,8 +6,11 @@
 *.sconsign.dblite
 *.org.eclipse.cdt.core.prefs
 *.language.settings.xml
+*.log
+*.jou
+*.project
 *.cproject
-
+WD-Firmware/demo/build/configure.txt
```
Comment (oferShinaar) [inline .gitignore]: change to configure.txt

File: WD-Firmware/demo/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+`Note: the attributes strName and listSconscript are mandatory ones. you should keep it in the class`
+
+* **strName** - *string*: will hold the name of the demo to be used throughout the build system
+* **listSconscripts** - *list*: will hold the Sconscripts that the user want to include in the demo's build
+
+Once done with setting up the demo, it will appear in the list shown to the user when running *config.sh* and the user can pick it up by entering it's index.
+
+The user may also add other attributes to the demo class if there is a need to. but these attributes must not put to use in shared sconscripts, so they will not break the build for other demos.
+
```
Comment (oferShinaar) [inline WD-Firmware/demo/README.md]: remove this section

File: WD-Firmware/psp/api_inc/psp_api.h [hunk 3]
```diff
@@ -39,6 +42,9 @@
 /**
 * macros
 */
+#define	M_PSP_CLR_TIMER_INT()      M_PSP_CLEAR_CSR(mie, D_PSP_MIP_MTIP);
```
Comment (oferShinaar) [inline WD-Firmware/psp/api_inc/psp_api.h]: align space

File: WD-Firmware/psp/api_inc/psp_attributes.h [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+/**
+* include files
+*/
+
+/**
+* definitions
+*/
+
+#ifdef __GNUC__  /* NatiR - to do - check for LLVM too */
```
Comment (oferShinaar) [inline WD-Firmware/psp/api_inc/psp_attributes.h]: do the "todo" we work with llvm

File: WD-Firmware/psp/api_inc/psp_intrinsics.h [hunk 5]
```diff
@@ -15,57 +15,32 @@
...
-* @date   21.01.2019 
-* @brief  The file implements the RTOS AL memory API
+* @file   psp_intrinsics.h
+* @author Nati Rapaport
+* @date   12.11.2019
+* @brief  The psp intrinsics
 * 
 */
+#ifndef  __PSP_INTRINSICS_H__
+#define  __PSP_INTRINSICS_H__
```
Comment (oferShinaar) [inline WD-Firmware/psp/api_inc/psp_intrinsics.h:26]: add define for clang and gnu

File: WD-Firmware/psp/src/psp_traps.c [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+* @brief  This file implements trap functions
+*
+*/
+
+/*-----------------------------------------------------------
+ * Implementation of functions defined in portable.h for the RISC-V RV32 port.
+ *----------------------------------------------------------*/
+#include "psp_types.h"
+#include "psp_api.h"
+
```
Comment (oferShinaar) [inline WD-Firmware/psp/src/psp_traps.c]: since we use "asm": add ifdef clang and GNU

File: WD-Firmware/demo/README.md (hunk 2 above)
Comment (nati-rapaport) [reply WD-Firmware/demo/README.md]: done

File: WD-Firmware/rtos/rtos_core/freertos/Source/portable/portASM.S [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+    engineered and independently SIL3 certified version for use in safety and
+    mission critical applications that require provable dependability.
+
+    1 tab == 4 spaces!
+*/
+
+/*
+This implementation supports riscv privileged v1.10
+*/
+#include "psp_core_base.inc"
```
Comment (oferShinaar) [inline WD-Firmware/rtos/rtos_core/freertos/Source/portable/portASM.S:81]: just a comment:  "base" is coming from base 32 or 64

File: WD-Firmware/rtos/rtos_core/freertos/Source/portable/portASM.S [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+    Integrity Systems ltd. to sell under the OpenRTOS brand.  Low cost OpenRTOS
+    licenses offer ticketed support, indemnification and commercial middleware.
+
+    http://www.SafeRTOS.com - High Integrity Systems also provide a safety
+    engineered and independently SIL3 certified version for use in safety and
+    mission critical applications that require provable dependability.
+
+    1 tab == 4 spaces!
+*/
+
```
Comment (oferShinaar) [inline WD-Firmware/rtos/rtos_core/freertos/Source/portable/portASM.S:72]: please add a comment: modify by WD (with your name) to support HiFive core with our FW-Infra

File: WD-Firmware/psp/api_inc/psp_attributes.h (hunk 4 above)
Comment (nati-rapaport) [reply WD-Firmware/psp/api_inc/psp_attributes.h]: done

File: WD-Firmware/rtos/rtos_core/freertos/Source/portable/portASM.S (hunk 7 above)
Comment (nati-rapaport) [reply WD-Firmware/rtos/rtos_core/freertos/Source/portable/portASM.S:81]: done

File: WD-Firmware/rtos/rtos_core/freertos/Source/portable/portASM.S (hunk 8 above)
Comment (nati-rapaport) [reply WD-Firmware/rtos/rtos_core/freertos/Source/portable/portASM.S:72]: done

File: .gitignore (hunk 1 above)
Comment (nati-rapaport) [reply .gitignore]: done and check whether the change is apropriate

Comment (nati-rapaport) [reply .gitignore]: done

File: WD-Firmware/psp/api_inc/psp_api.h (hunk 3 above)
Comment (nati-rapaport) [reply WD-Firmware/psp/api_inc/psp_api.h]: done

File: WD-Firmware/psp/api_inc/psp_intrinsics.h (hunk 5 above)
Comment (nati-rapaport) [reply WD-Firmware/psp/api_inc/psp_intrinsics.h:26]: done

File: WD-Firmware/psp/src/psp_traps.c (hunk 6 above)
Comment (nati-rapaport) [reply WD-Firmware/psp/src/psp_traps.c]: done

**Review by oferShinaar — APPROVED** [review]

**Pull request merged**